### PR TITLE
Bug 1189519 - Only download signatures for selected platform

### PR DESCRIPTION
**Pull request by MikeLing**
screen shot 2015-08-26 at 6 47 26 pm

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/914)

### CODE REVIEW

**Comment by MikeLing** [comment]
screen shot 2015-08-27 at 1 36 28 pm
Hmm, minor problem I think. You can see it haven't element been populated in the first line, but second time it will be fine to choose. Just don't know how to fix that.

**Comment by MikeLing** [comment]
Had address the problem in last comment :)

File: ui/js/graphs.js [hunk 1]
```diff
@@ -872,35 +872,35 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
...
-                                  _.remove($scope.unselectedTestList, {
-                                      projectName: test.projectName,
-                                      signature: test.signature });
-                              });
-                };
-
-                $scope.updateTestSelector();
-                $scope.loadingTestData = false;
+                $scope.selectedPlatform = $scope.platformList[0];
+                $scope.loadingPlatformList = false;
```
Comment (wlach) [inline ui/js/graphs.js]: I think you need to call updateTestSelector() here, as initially nothing is loaded until you select another platform.

File: ui/js/graphs.js [hunk 2]
```diff
@@ -872,35 +872,35 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
...
+        $scope.updateTestSelector = function() {
+            PhSeries.getSeriesByPlatform($scope.selectedProject.name,
+                $scope.timeRange, $scope.selectedPlatform, optionCollectionMap).then(
+                function(seriesData) {
+                    $scope.unselectedTestList = seriesData.seriesList;
+                }
+            );
+            // filter out tests which are already displayed or are
+            // already selected
+            _.forEach(_.union(testsDisplayed, $scope.testsToAdd),
```
Comment (wlach) [inline ui/js/graphs.js]: We lost the sorting of the unselected test list here, could you add it back? It's just a call to `_.sortBy`.

File: ui/js/perf.js [hunk 3]
```diff
@@ -59,6 +59,47 @@ perf.factory('PhSeries', ['$http', 'thServiceDomain', function($http, thServiceD
...
+        return $http.get(specifyPlatformURL).then(function(response) {
+            var seriesList = [];
+            var testList = [];
+
+            Object.keys(response.data).forEach(function(signature){
+                var seriesSummary = _getSeriesSummary(projectName, signature,
+                    response.data[signature], optionMap);
+                seriesList.push(seriesSummary);
+
+
```
Comment (wlach) [inline ui/js/perf.js]: Should only be one newline here.

File: ui/js/perf.js [hunk 4]
```diff
@@ -59,6 +59,47 @@ perf.factory('PhSeries', ['$http', 'thServiceDomain', function($http, thServiceD
...
+    var _getSeriesByPlatform = function(projectName, timeRange, platform, optionMap) {
+        var specifyPlatformURL = thServiceDomain + '/api/project/' + projectName +
+            '/performance-data/get_performance_series_summary/?interval=' +
+            timeRange + '&machine_platform=' + platform;
+
+        return $http.get(specifyPlatformURL).then(function(response) {
+            var seriesList = [];
+            var testList = [];
+
+            Object.keys(response.data).forEach(function(signature){
```
Comment (wlach) [inline ui/js/perf.js]: Space between function(signature) and {

Also, I prefer `_.keys(response.data)`, as it's a bit more concise.

File: ui/js/graphs.js [hunk 5]
```diff
@@ -872,35 +872,35 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
...
-                };
-
-                $scope.updateTestSelector();
-                $scope.loadingTestData = false;
+                $scope.selectedPlatform = $scope.platformList[0];
+                $scope.loadingPlatformList = false;
             });
+
+        $scope.updateTestSelector = function() {
+            PhSeries.getSeriesByPlatform($scope.selectedProject.name,
```
Comment (wlach) [inline ui/js/graphs.js]: I think you need to set $scope.loadingTestData to true here.

File: ui/js/graphs.js (hunk 2 above)
Comment (wlach) [inline ui/js/graphs.js]: Also, all this logic should happen just after we've set $scope.unselectedTestList, around line 890. Doing it this way means we're applying the logic on the old test data, which isn't what we want.

**Comment by wlach** [comment]
Merged manually in f2c918ec9

**Pull request closed without merging**